trl/models/utils.py

```diff
@@ -170,7 +172,13 @@ def add_hooks(model: "DeepSpeedEngine") -> None:
         optimizer_offload = model.optimizer.parameter_offload
     elif model.optimizer is not None:
         optimizer_offload = model.optimizer
-    optimizer_offload._register_hooks_recursively(optimizer_offload.module)
+    else:
+        raise NotImplementedError(f"Didn't yet handle optimizer {model.optimizer}.")
```

Comment:
In what case does it occur?

Reply:
Oh yeah basically I was getting IDE (PyCharm) warnings: "Local variable 'optimizer_offload' might be referenced before assignment"

> In what case does it occur?

It could happen if `model.optimizer is None`. That being said, this `else` branch definitely doesn't happen in practice, otherwise someone would have already reported this logic bug, as it'd lead to a `NameError`.

Feel free to ask me to revert this change, it just nice-ifies this failure branch if hit

Reply:
```suggestion
    else:
        raise RuntimeError("The model optimizer is None, which is not supported.")
```